mysql-test/suite/rpl/t/rpl_parallel_sbm.test

```diff
@@ -16,22 +17,42 @@
 # Seconds_Behind_Master is based on the timestamp of the new transaction,
 # rather than the last committed transaction.
 #
+#   Note that the test doesn't actually use the value of MASTER_DELAY, but
+# rather uses debug_sync to coordinate the end of the delay. This is to ensure
+# that on slow machines, a delay doesn't pass before the test gets a chance to
+# validate results. Additionally, it lets us continue testing after validation
+# so we don't have to wait out a full delay unnecessarily. The debug_sync point
+# is enabled via sql_delay_by_debug_sync, which will delay transactions based
+# only on GTID events, so only one synchronization is needed per transaction.
+#
 
 --connection slave
 --source include/stop_slave.inc
 --let $save_dbug= `SELECT @@GLOBAL.debug_dbug`
 --let $save_parallel_mode= `SELECT @@GLOBAL.slave_parallel_mode`
-set @@GLOBAL.debug_dbug= "d,negate_clock_diff_with_master";
 set @@GLOBAL.slave_parallel_mode= CONSERVATIVE;
---let $master_delay= 3
+set @@GLOBAL.debug_dbug= "d,negate_clock_diff_with_master,sql_delay_by_debug_sync";
+
+--let $master_delay= 1
 --eval change master to master_delay=$master_delay, master_use_gtid=Slave_Pos
 --source include/start_slave.inc
 
 --connection master
 --let insert_ctr= 0
 create table t1 (a int);
+--source include/save_master_gtid.inc
+--connection slave
+set debug_sync= "now WAIT_FOR at_sql_delay";
+set debug_sync= "now SIGNAL continue_sql_thread";
+--source include/sync_with_master_gtid.inc
+
+--connection master
 create table t2 (a int);
---source include/sync_slave_sql_with_master.inc
```

Comment:
.. as mtr was used to time out on this line?

Reply:
Then I did not get the idea of this synchronization conversion.
In the OLD version l.34 reads l.50 has been executed. There's no more to apply on slave.
The new version is different in that the GTID of the CREATE-TABLE gets in sync additionally which I don't see anyhow relevant to the rest of the test.